owncloudApp/src/androidTest/java/com/owncloud/android/settings/security/OCSettingsPasscode.kt

```diff
@@ -0,0 +1,226 @@
+        assertTrue(errorMessage, activityRule.activity.isFinishing)
+    }
+
+    @Test
+    fun secondTryIncorrect(){
+        //Open Activity in passcode creation mode
+        intent.setAction(PassCodeActivity.ACTION_REQUEST_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        //First typing
+        typePasscode(arrayOf('1','1','1','1'))
+        //Second typing
+        typePasscode(arrayOf('1','1','1','2'))
+
+        pressBack()
+
+        onView(withText(R.string.pass_code_reenter_your_pass_code)).check(doesNotExist())
+        onView(withText(R.string.pass_code_configure_your_pass_code)).check(matches(isDisplayed()))
+        onView(withText(R.string.pass_code_configure_your_pass_code_explanation)).check(matches(isDisplayed()))
+        onView(withText(R.string.pass_code_mismatch)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun cancelFirstTry() {
+        //Open Activity in passcode creation mode
+        intent.setAction(PassCodeActivity.ACTION_REQUEST_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        onView(withId(R.id.txt0)).perform(replaceText("1"))
+        onView(withId(R.id.txt1)).perform(replaceText("1"))
+        onView(withId(R.id.txt2)).perform(replaceText("1"))
+
+        onView(withId(R.id.cancel)).perform(click())
+        assertTrue(errorMessage, activityRule.activity.isFinishing)
+    }
+
+    @Test
+    fun cancelSecondTry() {
+        //Open Activity in passcode creation mode
+        intent.setAction(PassCodeActivity.ACTION_REQUEST_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        //First typing
+        typePasscode(arrayOf('1','1','1','1'))
+
+        onView(withId(R.id.txt0)).perform(replaceText("1"))
+        onView(withId(R.id.txt1)).perform(replaceText("1"))
+
+        onView(withId(R.id.cancel)).perform(click())
+        assertTrue(errorMessage, activityRule.activity.isFinishing)
+    }
+
+    @Test
+    fun deletePasscodeView() {
+        //Save a passcode in Preferences
+        storePasscode()
+
+        //Open Activity in passcode deletion mode
+        intent.setAction(PassCodeActivity.ACTION_CHECK_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        onView(withText(R.string.pass_code_remove_your_pass_code)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun deletePasscodeCorrect() {
+        //Save a passcode in Preferences
+        storePasscode("1111")
+
+        //Open Activity in passcode deletion mode
+        intent.setAction(PassCodeActivity.ACTION_CHECK_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        //Type correct passcode
+        typePasscode(arrayOf('1','1','1','1'))
+
+        assertTrue(errorMessage, activityRule.activity.isFinishing)
+    }
+
+    @Test
+    fun deletePasscodeIncorrect() {
+        //Save a passcode in Preferences
+        storePasscode("1111")
+
+        //Open Activity in passcode deletion mode
+        intent.setAction(PassCodeActivity.ACTION_CHECK_WITH_RESULT)
```

Comment:
And a new `openPasscodeCheck()` method with this line and the one below or **even a unique method with something like:**

```
private fun openPasscodeActivity (mode: string) {
    intent.setAction(mode)
    activityRule.launchActivity(intent)
} 
```

And you could use it to open the activity in pattern creation or deletion mode

Reply:
Done 👍